src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -198,6 +211,38 @@ public String createSnapshot(
         .submit();
   }
 
+  /**
+   * If the snapshot request is byRequestId, verify that the request has been approved and that a
+   * snapshot has not yet been successfully created from the request. Note: If the flightId is
+   * populated but the createdSnapshotId is not, then the previous flight failed and the snapshot
+   * creation should be allowed to continue.
+   *
+   * @param snapshotRequestModel to validate
+   */
+  @VisibleForTesting
+  void validateForByRequestIdMode(SnapshotRequestModel snapshotRequestModel) {
+    SnapshotRequestContentsModel requestContents = snapshotRequestModel.getContents().get(0);
+    SnapshotAccessRequestResponse snapshotAccessRequest =
+        snapshotRequestDao.getById(requestContents.getRequestIdSpec().getSnapshotRequestId());
+    if (snapshotAccessRequest.getStatus() != SnapshotAccessRequestStatus.APPROVED) {
+      throw new ValidationException(
+          "Snapshot request must be approved before creating a snapshot.");
+    }
+    if (snapshotAccessRequest.getCreatedSnapshotId() != null) {
+      throw new ValidationException(
+          "Snapshot with id %s is already created from request with id %s"
+              .formatted(
+                  snapshotAccessRequest.getCreatedSnapshotId(), snapshotAccessRequest.getId()));
+    }
+    if (snapshotAccessRequest.getFlightid() != null
+        && jobService.unauthRetrieveJobState(snapshotAccessRequest.getFlightid())
+            != FlightStatus.ERROR) {
+      throw new ValidationException(
+          "Snapshot Create Flight with id %s is still running"
+              .formatted(snapshotAccessRequest.getFlightid()));
+    }
```

Comment:
Might be a little clearer to declare the flight id once
```suggestion
    var flight = snapshotAccessRequest.getFlightid();
    if (flight != null && jobService.unauthRetrieveJobState(flight) != FlightStatus.ERROR) {
      throw new ValidationException(
          "Snapshot Create Flight with id %s is still running".formatted(flight));
    }
```